config/routes.rb

```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get 'users', to: 'users#index'
+  get '/users/:user_id', to: 'users#display'
+  get 'users/:user_id/posts', to: 'posts#index'
+  get 'users/:users_id/posts/:id', to: 'posts#display'
```

Comment:
- [ ] There seems to be a lot of inconsistency with your routing here. Sometimes you define `users_id`, other times you define `user_id`. Rails takes care of these routes for us. We can nest these routes and also use the `resources` helper from Rails.

  ```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
  ```

Reply:
Thank you for your help @TedLivist 
I appreciate